crates/shim/src/lib.rs

```diff
@@ -71,13 +72,19 @@ pub mod api {
     pub use super::protos::types::empty::Empty;
 }
 mod args;
+pub mod common;
 mod logger;
 pub mod monitor;
 pub mod mount;
 mod publisher;
 mod reap;
 pub mod util;
 
+#[cfg(feature = "async")]
```

Comment:
Can we do it mutually exclusive? e.g. don't allow sync API when `async` enabled (I think its ok to define async api side by side with sync ones and use feature toggle to exclude).

Reply:
Now the async and sync specific codes are splited into `synchronous` and `asynchronous` folder `shim` and `runc-shim`. pls check